public/js/StatusBar.js

```diff
@@ -0,0 +1,194 @@
+        //toggle flash critical state every time interval
+        if (this.secondsBeforeFlash <= 0) {
+            this.secondsBeforeFlash = sbc.STATUS_BAR_CRITICAL_FLASH_INTERVAL;
+            if (this.flashCritical)
+                this.flashCritical = false;
+            else 
+                this.flashCritical = true;
+        }
+    }
+
+    draw() {   
+        super.draw();
+        var ctx = this.game.ctx;
+
+        var colorForeground = this.foregroundColor;
+        var colorBackground = this.backgroundColor;
+        if (this.hasCriticalState && this.current <= sbc.STATUS_BAR_CRITICAL_AMOUNT * this.maxSize) {
+            //paint the critical state
+            if (this.flashCritical) {
+                colorForeground = this.criticalForegroundColor;
+                colorBackground = this.criticalBackgroundColor;
+            }
+
+        }
+
+        //draw the background status bar
+        ctx.save();
+        ctx.lineWidth = sbc.STATUS_BAR_WIDTH;
+        ctx.strokeStyle = colorBackground;
+        ctx.lineCap = "round";
+        ctx.beginPath();
+        ctx.moveTo(this.x, this.y);
+        ctx.lineTo(this.x + this.maxLength, this.y);
+        ctx.stroke();
+        ctx.closePath();
+        
+        if (this.current > 0) {
+            //draw the foreground status bar over the background status bar
+            ctx.strokeStyle = colorForeground;
+            //ctx.lineCap = butt;
+            ctx.beginPath();
+            ctx.moveTo(this.x, this.y);
+            ctx.lineTo(this.x + ((this.maxLength / this.maxSize) * this.current), this.y);
+            ctx.stroke();
+            ctx.closePath();
+
+        }
+
+        //draw the text of the status bar current/maxSize
+        if (this.displayText) {
+            ctx.fillText(`${this.current} / ${this.maxSize}`, this.x + this.maxLength/2, 
+                this.y + sbc.STATUS_BAR_WIDTH * 0.20);
+        }
+        
+        ctx.restore();
+
+        
+    }
+
+    /*
+    * This method will be overwritten by the child classes and will set the current value 
+    * of the status bar every update. for example, a status bar could set the current value to
+    * an entities health.
+    */
+    setCurrent() {
+        throw new Error("AbstractStatusBar can't be instantiated");
+    }
+
+    //set the max size of the status bar, allows dynamic changing of the max size of the bar
+    setMaxSize() {
+        throw new Error("AbstractStatusBar can't be instantiated");
+    }
+}
+
+
+/*
+* This class is the health status bar.
+* This bar will keep track of zerlins health and display it for the
+* player to see the a visual representation of the current health.
+*/ 
+class HealthStatusBar extends AbstractStatusBar {
+    constructor(game, x, y) {
+        super(game, x, y, sbc.HEALTH_BAR_HAS_CRITICAL_STATE);
+        this.foregroundColor = "rgba(255, 0, 0, 1)"; //red
+        this.backgroundColor = "rgba(255, 126, 126, 1)"; //light red
```

Comment:
What do you think about setting a text color here to be more readable for against the bar color?